File: src/main/java/page/awt/BBBuilder.java
package page.awt;

import common.battle.BattleField;
import common.battle.SBCtrl;
import page.anim.IconBox;
import page.battle.BattleBox;
import page.battle.BattleBox.OuterBox;
import page.view.ViewBox;

public abstract class BBBuilder {

	public static BBBuilder def;

	public abstract BattleBox getCtrl(OuterBox bip, SBCtrl bf);

	public abstract BattleBox getDef(OuterBox bip, BattleField bf);

	public abstract IconBox getIconBox();

	public abstract BattleBox getRply(OuterBox bip, BattleField bf, String str, boolean t);

	public abstract ViewBox getViewBox();

}

File: src/page/awt/BBBuilder.kt
package page.awt

import common.battle.BattleField
import common.battle.SBCtrl
import page.anim.IconBox
import page.battle.BattleBox
import page.battle.BattleBox.OuterBox
import page.view.ViewBox

abstract class BBBuilder {
    abstract fun getCtrl(bip: OuterBox?, bf: SBCtrl?): BattleBox
    abstract fun getDef(bip: OuterBox?, bf: BattleField?): BattleBox
    abstract val iconBox: IconBox
    abstract fun getRply(bip: OuterBox?, bf: BattleField?, str: String, t: Boolean): BattleBox
    abstract val viewBox: ViewBox

    companion object {
        var def: BBBuilder? = null
    }
}
